jsonschema/cli.py

```diff
@@ -17,31 +110,39 @@ def _namedAnyWithDefault(name):
     return namedAny(name)
 
 
-def _json_file(path):
-    with open(path) as file:
-        return json.load(file)
-
-
 parser = argparse.ArgumentParser(
     description="JSON Schema Validation CLI",
 )
 parser.add_argument(
     "-i", "--instance",
     action="append",
     dest="instances",
-    type=_json_file,
+    type=str,
     help=(
-        "a path to a JSON instance (i.e. filename.json) "
-        "to validate (may be specified multiple times)"
+        "A path to a JSON instance (i.e. filename.json) "
+        "to validate (may be specified multiple times). "
+        "If this option is not used, an instance can be given through stdin. "
+        "If neither -i nor stdin is given, only the schema is checked."
```

Comment:
This behavior seems true (that if you don't provide an instance the schema is still validated), but I'm not sure it was fully intentional.

It's sensical certainly, so I'm happy definitely documenting it, but I think adding a test to make sure it stays the case would probably be smart.

Reply:
Added in `test_schema_validation`

Reply:
Great, thanks!